databricks/koalas/frame.py

```diff
@@ -2896,11 +2896,11 @@ def fillna(self, value=None, method=None, axis=None, inplace=False, limit=None):
         We can also propagate non-null values forward or backward.
 
         >>> df.fillna(method='ffill')
-             A    B    C  D
-        0  NaN  2.0  NaN  0
-        1  3.0  4.0  NaN  1
-        2  3.0  4.0  NaN  5
-        3  3.0  3.0  1.0  4
+             A    B    C    D
+        0  NaN  2.0  NaN  0.0
```

Comment:
@RainFung, BTW, how does it work with other types like strings in pandas? seems like we cast into floats always.

Reply:
Also seems like pandas doesn't change the type for the column `D`.

E.g., with a string column `E`:

```py
>>> import pandas as pd
>>> df = pd.DataFrame({'A': [None, 3, None, None], 'B': [2, 4, None, 3], 'C': [None, None, None, 1], 'D': [0, 1, 5, 4], 'E': [None, 'B', None, 'D']})
>>> df
     A    B    C  D     E
0  NaN  2.0  NaN  0  None
1  3.0  4.0  NaN  1     B
2  NaN  NaN  NaN  5  None
3  NaN  3.0  1.0  4     D

>>> df.fillna(method='ffill')
     A    B    C  D     E
0  NaN  2.0  NaN  0  None
1  3.0  4.0  NaN  1     B
2  3.0  4.0  NaN  5     B
3  3.0  3.0  1.0  4     D
```